pybamm/models/base_model.py

```diff
@@ -226,6 +232,14 @@ def jacobian(self):
     def jacobian(self, jacobian):
         self._jacobian = jacobian
 
+    @property
+    def jacobian_algebraic(self):
```

Comment:
why is there a jacobian_algebraic but no jacobian_rhs?

Reply:
only because we only ever use either the full jacobian (for the solver) or the algebraic part (for consistent ics). I'll add `jacobian_rhs` too for completeness.

Reply:
oh right, I had forgotten about that, probably fine without `jacobian_rhs` then. Need to make sure we aren't calculating the jacobian for the algebraic part twice then (can't remember if already doing this)